h/static/scripts/directive/annotation.coffee

```diff
@@ -202,6 +208,10 @@ AnnotationController = [
       switch @action
         when 'create'
           updateDomainModel(model, @annotation)
+
+          if group.focusedGroup().hashid != '__public__'
```

Comment:
We shouldn't set it. Server-side, an annotation that belongs to the "Public" group is an annotation that has no group field. Client-side this is represented as belong to a group called "Public" with hashid "__public__" but that's only because Public looks and functions like a group in the UI (in the groups dropdown etc), it has no meaning server-side (and in fact the server wouldn't let you create an annotation in this group since the group doesn't exist and you aren't auth'd to write to it) so the client strips it before sending the annotation.

There's some explanation of why this was done (and what other possibilities were considered and why they were rejected) in this commit message: <https://github.com/hypothesis/h/commit/ef1dc7741310d55783247f323ac99cd6957b20fb>:

> - Annotations that don't belong to any group / that are public simply have no
>   `group` field.
> 
>   Using `group: "public"` (actually it would have to be `__public__` so not to
>  clash with any generated hashid) and `group: "__none__"` were considered. Of
>   these I prefer `none` because it doesn't imply the existence of a "public
>   group" that doesn't really exist, it's just a sentinel value used to mark
>   annotations with no group. The advantage was that every annotation always
>   having a group field with a string value is more consistent and would make
>   coding (both h code and API client code) simpler. But:
> 
>   1. We would have to write an Elasticsearch migration script to add `group:
>   "__none__"` to all pre-existing annotations.
> 
>   2. We'd have to add code to the API to ensure that even if the client sends
>   no group field when creating an annotation, `group: "__none__"` gets
>   inserted.
> 
>   3. You still have to write code to handle annotations with group `"__none__"`
>   differently all over the place.
> 
>   I decided it was simpler just to write code to handle annotations with no
>   group field.